### DISCUSSION TITLE
graphql-env.d.ts doesn't generate

### DISCUSSION THREAD

**Discussion by aadamsx** [0no-co/gql.tada#106] [Q&A]
I have a regualr typescript node server.  I've been using Restful APIs up to this point but have decided to try out graphql.  I ran across gql.tada and wanted to give it a shot.  I'm using VSCode.

installed:

```bash
npm install gql.tada
npm install --save-dev @0no-co/graphqlsp
```

updated my tsconfig.json file:

```json
    "plugins": [
      {
        "name": "@0no-co/graphqlsp",
        "schema": "./schema.graphql",
        "tadaOutputLocation": "./src/graphql-env.d.ts"
      }
    ],

 ```
    
updated vscode settings (I'm using a workspace with several projects so specified the ```server``` in the path):
    
 ```json
{
  "typescript.tsdk": "server/node_modules/typescript/lib",
  "typescript.enablePromptUseWorkspaceTsdk": true
}

```

created a ```schema.graphql``` in root with my types.  I also installed GraphQL Language Feature Support and Syntax Highlighting.  

I then restarted my server thinking that a ```graphql-env.d.ts``` file would be created in my ```server/src/``` directory, but it never showed.  So I created an empty one, and it never got filled in.  

I thought maybe I needed to run my script to get the ```graphql-env.d.ts``` file to generate:

```json
"dev/server": "nodemon --watch 'src/**/*.ts' --exec 'ts-node -r tsconfig-paths/register' src/server.ts",
```

```bash
npm run dev/server
```

but this did nothing to generate the file.

I'm using the following version:

```json

  "dependencies": {
    "gql.tada": "^1.2.1",
  },
  "devDependencies": {
    "@0no-co/graphqlsp": "^1.4.2",
    "@types/node": "^20.10.7",
    "@typescript-eslint/eslint-plugin": "^6.18.1",
    "@typescript-eslint/parser": "^6.18.1",
    "eslint": "^8.56.0",
    "eslint-config-xo": "^0.43.1",
    "eslint-config-xo-typescript": "^1.0.1",
    "nodemon": "^3.0.2",
    "ts-node": "^10.9.2",
    "tsc-alias": "^1.8.8",
    "tsconfig-paths": "^4.2.0",
    "typescript": "^5.3.3"
  }
```

What could I be doing wrong?

**Comment by maraisr** [comment]
Does https://github.com/0no-co/gql.tada/discussions/49#discussioncomment-8337310 answer your question?

**Comment by aadamsx** [comment]
I selected the Use Workspace Version:

<img width="737" alt="image" src="https://github.com/0no-co/gql.tada/assets/5341158/d6b08d20-4753-4dda-bbe5-d7dac0977aa1">

Then opened a ts file 

<img width="1079" alt="image" src="https://github.com/0no-co/gql.tada/assets/5341158/9e3d1c7e-5939-4231-a02d-8c8c6e5d88b1">


referenced in my```include``` via tsconfig.json source in the VSCode editor.  

<img width="335" alt="image" src="https://github.com/0no-co/gql.tada/assets/5341158/4cf45b7e-0874-45fb-9af7-32bf8970a3ab">

And restarted my VCode instance.  No ```graphql-env.d.ts``` file was generated.  That was the main gist of comment #49 I believe.

I turned on TSLogging, ```normal``` mode.  

<img width="1127" alt="image" src="https://github.com/0no-co/gql.tada/assets/5341158/d89047fe-c383-4114-a831-4a70fd5834eb">


I searched for ```GraphQLSP``` in the logs and didn't see any results.

<img width="2115" alt="image" src="https://github.com/0no-co/gql.tada/assets/5341158/ac8ea0be-9301-4d54-bbbf-1147428977ca">

**Reply by kitten** [reply]
It's hard to debug this without having a reproduction. You may be able to figure out what's wrong by running the `gql.tada doctor` command. But it could be an issue with which `tsconfig.json` is being loaded or how it's set up.
